hack/lib/util.sh

```diff
@@ -853,6 +853,23 @@ function kube::util::ensure_dockerized {
   fi
 }
 
+# kube::util::ensure-gnu-sed
+# Determines which sed binary is gnu-sed on linux/darwin
+#
+# Sets:
+#  SED: The name of the gnu-sed binary
+#
+function kube::util::ensure-gnu-sed {
```

Comment:
looks fine for me.

Another more radical idea: just define a function `sed` pointing to the right binary. I.e. not global var, just overloading the sed command. Automatically in init.sh.

Reply:
That is a bit radical! The upside is it’s just there and you don’t have to think about it. The downside is that it’s obscuring what it’s doing from the person writing the script, which makes it harder to diagnose. Possible unintended consequences.